### PR TITLE
create iCalendar file

### LINKED ISSUE #14: สร้างไฟล์ ICS (iCalendar file)
เอาข้อมูล JSON มาแปลงเป็นไฟล์ [iCalendar](https://en.wikipedia.org/wiki/ICalendar) จะได้ใช้กับพวกแอพ Calendar ได้

## Example of apps supporting ICS URLs

อย่างเช่นแอพ

| App | Example |
| --- | --- |
| Google Calendar | image |
| Calendar (macOS) | image |

## Implementation Suggestion

โปรเจกต์นี้ใช้ Node.js + Yarn ให้โคลนโปรเจกต์ แล้วสั่ง `yarn install` เพื่อติดตั้งลงไลบรารี่ต่างๆ ครับ
หลังจากนั้นสั่ง `yarn build` จะได้ไฟล์ `public/calendar.json`

แนะนำให้สร้าง Script ใหม่ไว้ที่ไฟล์ `scripts/generate-ics.js`
พอรันแล้วให้มันอ่านไฟล์ `public/calendar.json` แล้วเอามาสร้างเป็นไฟล์ `public/calendar.ics` ครับ

แนะนำให้ดูตัวอย่าง Script ที่ใกล้เคียง
คือ Script เอาไว้[สร้างรูปปฏิทิน](https://github.com/ThaiProgrammer/tech-events-calendar#tech-events-calendar) ที่ [scripts/generate-svg.js](https://github.com/ThaiProgrammer/tech-events-calendar/blob/master/scripts/generate-svg.js) ครับ

**หากต้องการทำ Issue นี้ ช่วยคอมเม้นต์ไว้ด้วยนะครับ จะได้ไม่ทำงานซ้ำซ้อนกัน**

### PR DESCRIPTION
**Pull request by koobitor**
close #14 
- Create script scripts/generate-ics.js
- ignore /public/calendar.ics
- add script node scripts/generate-ics.js to package.json

### CODE REVIEW

File: scripts/generate-ics.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        endDate = new Date(event.end.year, event.end.month - 1, event.end.date + 1)
+        start = `DTSTART;VALUE=DATE:${toISODateString(startDate)}`
+        end = `DTEND;VALUE=DATE:${toISODateString(endDate)}`
+      }
+
+      const description = addNewlines(event.description.replace(/(?:\n)/g, '\\n')).trim()
+      
+      result.push([
+        `BEGIN:VEVENT\r`,
+        `UID:${guid()}`,
```
Comment (dtinth) [inline scripts/generate-ics.js]: I think UID should be deterministic — e.g. running it multiple times, same event should result in the same ID. So apps can properly track when events are added, changed, moved, or deleted.

File: scripts/generate-ics.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+function generateICS () {
+  const ics = [
+    `BEGIN:VCALENDAR`,
+    `VERSION:2.0`,
+    `PRODID:-//Thai Programmer//NONSGML v1.0//EN`,
+    `CALSCALE:GREGORIAN`,
+    `METHOD:PUBLISH`,
+    `${renderCells()}`,
+    `END:VCALENDAR`
+  ].join('\n')
```
Comment (dtinth) [inline scripts/generate-ics.js]: I think ICS files always use `\r\n`.

File: scripts/generate-ics.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      
+      result.push([
+        `BEGIN:VEVENT\r`,
+        `UID:${guid()}`,
+        `DTSTAMP:${toISOString(genDate)}`,
+        `${start}`,
+        `${end}`,
+        `CREATED:${toISOString(genDate)}`,
+        `SUMMARY:${event.title}`,
+        `LOCATION:${event.location.title}`,
```
Comment (dtinth) [inline scripts/generate-ics.js]: Maybe have to do more escaping on this one, according to RFC escaping rules (e.g. `,` must be escaped to `\,`). Or we can also try using https://github.com/tritech/node-icalendar.

File: scripts/generate-ics.js (hunk 1 above)
Comment (dtinth) [reply scripts/generate-ics.js]: I think to really do this, each event needs a unique ID.

**Comment by koobitor** [comment]
Thank you for great review.

**Pull request merged**